src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,90 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    List<String> tests = new ArrayList<String>();
+    String testngXmlAsString;
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {
+        verifyTestngXmlExists();
+        readTestngAsString();
+        
+        getTestFiles();
+        excludeFilesNotInTestng();
+
+        verifyTestngContainsTests();
+    }
+    
+    private void verifyTestngXmlExists() {
+        File f = new File("./src/test/testng.xml");   
+        assertTrue(f.exists() && !f.isDirectory());
+    }
+    
+    private void readTestngAsString() throws FileNotFoundException {
+        testngXmlAsString = FileHelper.readFile("./src/test/testng.xml");
+    }
+
+    /**
+     * Files to be checked in testng.xml are stored in tests
+     */
+    private void getTestFiles() {
+        getTestFiles("./src/test/java/teammates/test/cases", true); // BaseComponentTestCase, BaseTestCase excluded
+    }
+    
+    private void excludeFilesNotInTestng() {
+        tests.remove("BaseUiTestCase");
+        tests.remove("FeedbackQuestionUiTest");
+        tests.remove("GodModeTest");
+    }
+
+    private void verifyTestngContainsTests() {
+        for (String test : tests) {
+            assertTrue(testngXmlAsString.contains(test));
```

Comment:
This comparison is very weak. See if you can compare it with `<class name="teammates.test.cases.package.name.TestClassName" />` instead.